screens/ReservationViewScreen.js

```diff
@@ -0,0 +1,630 @@
+import React, {Component} from 'react'
+import {Field, reduxForm} from 'redux-form'
+import {Keyboard, Text, TouchableOpacity, View, FlatList, Dimensions, Alert, Animated, RefreshControl} from 'react-native'
+import {connect} from 'react-redux'
+import {compose} from "redux";
+import {withContext} from "../helpers/contextHelper";
+import {getTableLayouts, getReservation} from '../actions'
+import {getInitialTablePosition, getTablePosition, getSetPosition} from "../helpers/tableAction";
+import {api, dispatchFetchRequest, dispatchFetchRequestWithOption} from '../constants/Backend'
+import {LocaleContext} from '../locales/LocaleContext'
+import ScreenHeader from "../components/ScreenHeader";
+import {NavigationEvents} from "react-navigation";
+import LoadingScreen from "./LoadingScreen";
+import TimeZoneService from "../helpers/TimeZoneService";
+import styles from '../styles'
+import {StyledText} from "../components/StyledText";
+import {ThemeContainer} from "../components/ThemeContainer";
+import {ThemeScrollView} from "../components/ThemeScrollView";
+import moment from 'moment-timezone'
+import Ionicons from 'react-native-vector-icons/Ionicons';
+import {ThemeKeyboardAwareScrollView} from "../components/ThemeKeyboardAwareScrollView";
+
+
+class ReservationViewForm extends React.Component {
+  static navigationOptions = {
+    header: null
+  }
+  static contextType = LocaleContext
+
+  constructor(props, context) {
+    super(props, context)
+
+    this.state = {
+      windowWidth: Dimensions.get('window').width - 30,
+      windowHeight: Dimensions.get('window').height - 76,
+      scaleMultiple: (Dimensions.get('window').width - 30) / 300,
+      tableWidth: null,
+      tableHeight: null,
+      isTablet: context?.isTablet,
+      tableIndex: 0,
+      selectedTableIds: [],
+      selectedTableNames: [],
+      activeTableLayout: [],
+    }
+  }
+
+  componentDidMount() {
+
+    this.props.getTableLayouts()
+    this.props.getReservation()
+
+    this.context.localize({
+      en: {
+        noTableLayout:
+          'You need to define at least one table layout and one table.',
+        tableCapacity: 'Tables',
+        availableSeats: 'Vacant',
+        availableTables: 'Vacant',
+      },
+      zh: {
+        noTableLayout: '需要創建至少一個桌面跟一個桌位.',
+        tableCapacity: '總桌數',
+        availableSeats: '空位',
+        availableTables: '空桌',
+      }
+    })
+  }
```

Comment:
Is this needed?

Reply:
I think I can try to organize these to root file now, it was used in serval screens